### PR TITLE
Added the ability to toggle between light and dark themes.

### PR DESCRIPTION
**Pull request by cdterry87**

### CODE REVIEW

**Comment by cdterry87** [comment]
This should take care of Issue #56

File: src/store/modules/settings.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import gitlab from "@/gitlab";
 
 const state = {
+  darkMode: localStorage.getItem("darkMode"),
```
Comment (chamerling) [inline src/store/modules/settings.js]: Maybe you can transform the localstorage in boolean directly in the state?

File: src/App.vue [hunk 2]
```diff
@@ -177,6 +177,9 @@ export default {
     }
   },
   computed: {
+    darkMode() {
+      return (typeof this.$store.state.settings.darkMode != "boolean" ? true : this.$store.state.settings.darkMode);
```
Comment (chamerling) [inline src/App.vue]: What about? 

```suggestion
  return !!this.$store.state.settings.darkMode;
```

File: src/views/Settings.vue [hunk 3]
```diff
@@ -118,13 +135,22 @@ export default {
...
+      }
+    },
     desktopNotification() {
       return this.$store.state.notification.enabled;
     }
   },
   mounted() {
     this.localApiToken = this.$store.state.settings.apiToken;
     this.localApiEndpoint = this.$store.state.settings.apiEndpoint;
+    this.localDarkMode = (this.$store.state.settings.darkMode == "false" ? false : true);
```
Comment (chamerling) [inline src/views/Settings.vue]: Use boolean directly from the state

File: src/views/Settings.vue [hunk 4]
```diff
@@ -118,13 +135,22 @@ export default {
         this.localApiEndpoint = value;
       }
     },
+    darkMode: {
+      get() {
+        return this.$store.state.settings.darkMode;
+      },
+      set(value) {
+        this.localDarkMode = value;
```
Comment (chamerling) [inline src/views/Settings.vue]: You can avoid watching and dispatch directly in the setter

```suggestion
        this.$store.dispatch("updateTheme", value);
```

**Comment by chamerling** [comment]
There are components which are still dark

Capture d’écran 2019-10-04 à 10 15 38
Capture d’écran 2019-10-04 à 10 15 17
Capture d’écran 2019-10-04 à 10 15 12
Capture d’écran 2019-10-04 à 10 15 07
Capture d’écran 2019-10-04 à 10 14 58

**Comment by cdterry87** [comment]
@chamerling Thank you for letting me know.  I'll try working on getting these components today to make them reflect the specified theme, as well.

**Comment by cdterry87** [comment]
@chamerling I have made the requested changes.  Please let me know if there are any additional issues.

**Comment by chamerling** [comment]
@cdterry87 nice. Looks like there is on issue when selecting a team

Capture d’écran 2019-10-04 à 18 56 38

**Comment by cdterry87** [comment]
I must have overlooked that one.  I just submitted another commit to fix it.  Check it out now, and let me know if it is better now.

**Comment by chamerling** [comment]
This is great now :) Thx a lot!

**Review by chamerling — APPROVED** [review]

File: src/store/modules/settings.js (hunk 1 above)
Comment (chamerling) [reply src/store/modules/settings.js]: Can you address this one? Having the watcher is useless then

File: src/views/Settings.vue [hunk 5]
```diff
@@ -140,6 +168,12 @@ export default {
       this.$store.dispatch("cleanAll");
       this.localApiToken = this.$store.state.apiToken;
       this.localApiEndpoint = this.$store.state.apiEndpoint;
+      this.localDarkMode = this.$store.state.darkMode;
+    }
+  },
+  watch: {
+    localDarkMode(toggle) {
+      this.$store.dispatch("updateTheme", toggle);
```
Comment (chamerling) [inline src/views/Settings.vue]: Remove this useless watcher and dispatch in the setter has described above, WDYT?

**Review by chamerling — CHANGES_REQUESTED** [review]
Please remove the watcher, use the setter to dispatch the action instead

**Comment by cdterry87** [comment]
> Please remove the watcher, use the setter to dispatch the action instead

I attempted to resolve this a few different ways without the watcher, but using the watcher was the only way I could get it to work.  Any ideas? Could you point me in the right direction?

File: src/views/Settings.vue [hunk 6]
```diff
@@ -55,6 +55,22 @@
...
+          <v-card class="elevation-12 mt-5">
+            <v-toolbar dark>
+              <v-toolbar-title class="white--text">Theme</v-toolbar-title>
+            </v-toolbar>
+            <v-card-text>
+              <div class="body-1 mb-3">
+                Select light or dark mode.
+                <v-switch
+                  color="primary"
+                  v-model="localDarkMode"
```
Comment (chamerling) [inline src/views/Settings.vue]: bind to `darkMode` and remove ALL the `localDarkMode`declaration and references. The `darkMode` getter and setter should do the trick

**Comment by chamerling** [comment]
> > Please remove the watcher, use the setter to dispatch the action instead
> 
> I attempted to resolve this a few different ways without the watcher, but using the watcher was the only way I could get it to work. Any ideas? Could you point me in the right direction?

Check this https://github.com/chamerling/gitlab-teams/pull/60#discussion_r331870457

**Comment by cdterry87** [comment]
@chamerling lol, I was so close!  I have implemented your suggested changes.  Once again, I appreciate your help.  Let me know if there is anything else that needs to be done.

File: src/views/Settings.vue [hunk 7]
```diff
@@ -118,13 +134,22 @@ export default {
...
+      }
+    },
     desktopNotification() {
       return this.$store.state.notification.enabled;
     }
   },
   mounted() {
     this.localApiToken = this.$store.state.settings.apiToken;
     this.localApiEndpoint = this.$store.state.settings.apiEndpoint;
+    this.darkMode = this.$store.state.settings.darkMode;
```
Comment (chamerling) [inline src/views/Settings.vue]: This line is useless if you have the computed property

File: src/views/Settings.vue [hunk 8]
```diff
@@ -140,6 +165,12 @@ export default {
       this.$store.dispatch("cleanAll");
       this.localApiToken = this.$store.state.apiToken;
       this.localApiEndpoint = this.$store.state.apiEndpoint;
+      this.darkMode = this.$store.state.darkMode;
+    }
+  },
+  watch: {
+    darkMode(toggle) {
+      this.$store.dispatch("updateTheme", toggle);
```
Comment (chamerling) [inline src/views/Settings.vue]: Remove the watcher since you have the computed property

**Comment by chamerling** [comment]
@cdterry87 you are close to the perfection, I commented what you need to remove

**Comment by cdterry87** [comment]
@chamerling Sorry about that! The unnecessary lines have been removed.

**Pull request merged**